tests/marker_test.go

```diff
@@ -60,4 +67,47 @@ var _ = Describe("bridge-marker", func() {
 			}, 20*time.Second, 5*time.Second).Should(Equal(false))
 		})
 	})
+
+	Describe("pod requiring bridge resource", func() {
+		It("should be started only when bridge is available on node", func() {
+
+			resourceName := tests.GenerateResourceName()
+			requiredResourceCount := "1"
+			requiredResources := corev1.ResourceList{
+				resourceName: resource.MustParse(requiredResourceCount),
+			}
+
+			podReq := tests.PodSpec(TestPodName, requiredResources)
+
+			_, err := clientset.CoreV1().Pods(DefaultNamespace).Create(podReq)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				pod, err := clientset.CoreV1().Pods(DefaultNamespace).Get(TestPodName, v1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return pod.Status.Phase == "Pending"
+			}, 20*time.Second, 5*time.Second).Should(Equal(true))
+
+			// Pod should be stuck in "Pending", wait 10 seconds to see if it stays there
+			time.Sleep(10 * time.Second)
+
+			Eventually(func() bool {
+				pod, err := clientset.CoreV1().Pods(DefaultNamespace).Get(TestPodName, v1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return pod.Status.Phase == "Pending"
+			}, 20*time.Second, 5*time.Second).Should(Equal(true))
+
+			node, err := tests.AddBridgeOnSchedulableNode(clientset, tests.TestBridgeName)
+
+			Eventually(func() bool {
+				pod, err := clientset.CoreV1().Pods("default").Get(TestPodName, v1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return pod.Status.Phase == "Running"
+			}, 20*time.Second, 5*time.Second).Should(Equal(true))
```

Comment:
can you please add a check here that the pod is running on the same node that you get from the `node, err := tests.AddBridgeOnSchedulableNode(clientset, tests.TestBridgeName)` function.

it should be in the pod status struct